docs/dev/development/attestation-internals.rst

```diff
@@ -0,0 +1,365 @@
+Internals and Technical Details for PEP 740 on PyPI
+===================================================
+
+This page documents some of the internals and technical details behind
+PyPI's implementation of :pep:`740`.
+
+.. important::
+
+  If you're a user of PyPI, you probably want the `attestation user docs`_
+  instead.
+
+Signing identities
+------------------
+
+A signing identity is a stable, human-readable identifier associated with a
+public key. This identifier is used to perform semantic mappings for the
+purpose of verification, e.g. to say "Alice signed for ``foo``," rather than
+"Key ``0x1234...`` signed for ``foo``."
+
+In traditional signing schemes, this is typically a "key identifier,"
+such as a truncated hash of the key itself. In X.509-based PKIs it can be
+the certificate's subject or other identifying material (such as a domain
+name or email address).
+
+As specified in PEP 740, signing identities for attestations are
+*Trusted Publisher* identities. In practice, this means that the identity
+expected to sign a distribution's attestation is expected to match the
+Trusted Publisher that published the package.
+
+For example, for a GitHub-based Trusted Publisher, the identity might be
+``https://github.com/pypa/sampleproject/blob/main/.github/workflows/release.yml``,
+i.e. ``pypa/sampleproject`` on GitHub, publishing from a workflow defined
+on the ``main`` branch in the file ``release.yml``.
+
+.. _future-identities:
+
+Future identities
+^^^^^^^^^^^^^^^^^
+
+In the future, PyPI may allow signing identities other than the project's
+Trusted Publishers. Some potential future signing identities include:
+
+* E-mail addresses, checked against the project's owners' profiles
+  and/or against the uploaded distribution's own metadata.
+* Third-party identities (such as GitHub or GitLab usernames), checked against
+  the uploaded distribution's own metadata.
+
```

Comment:
```suggestion
```
I think we should avoid including forward-looking details like this in our docs.

Reply:
Makes sense, removed!